packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -500,12 +503,19 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   TextSelectionOverlay _selectionOverlay;
 
   final ScrollController _scrollController = ScrollController();
+  AnimationController _cursorController;
+
   final LayerLink _layerLink = LayerLink();
   bool _didAutoFocus = false;
 
+  static const Duration _fadeDuration = Duration(milliseconds: 250);
+  final bool _platformIsIOS = defaultTargetPlatform == TargetPlatform.iOS;
```

Comment:
This is still in the widgets layer. Instead of self-determining this, its widget's constructor should just have a bool arg that asks should the cursor blink be animated.